src/npc.cpp

```diff
@@ -2301,6 +2338,14 @@ void npc::on_load()
     if( g->m.veh_at( pos() ).part_with_feature( VPFLAG_BOARDABLE, true ) && !in_vehicle ) {
         g->m.board_vehicle( pos(), this );
     }
+    if( has_effect( effect_riding ) && !mounted_creature ) {
+        if( g->critter_at<monster>( pos() ) ) {
```

Comment:
```suggestion
        if( const monster *const mon = g->critter_at<monster>( pos() ) ) {
            mounted_creature = g->shared_from( *mon );
```
No need to drag a dependency on the creature_tracker in.